src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -179,21 +185,30 @@ def process(self, local_path, metadata, data=None):
         """
 
         if self._config_is_modified() or self.asapo_worker is None:
-            self.config.update(utils.load_config(self.config["user_config_path"]))
-            self.asapo_worker = AsapoWorker(self.config)
+            config = self._create_asapo_config()
+            self.asapo_worker = AsapoWorker(**config)
 
         self.asapo_worker.send_message(local_path, metadata)
 
+    def _create_asapo_config(self):
+        config = {k: v for k, v in self.config.items() if k != "user_config_path"}
```

Comment:
How about to this in `Plugin.__init__` instead:
```
self.user_config_path = config.pop("user_config_path")
self.config = config
```
Maybe `user_config_path` doesn't need to be configurable at all. I think one could get the parent of `current` from the `metadata` argument in `process`.

Reply:
What is inside of metadata and what is parent of current?

Reply:
For example:
```
local_path = "/path/to/current/raw/subdir/filename.ext"
metadata["relative_path"] = "current/raw/subdir"
metadata["filename"] = "filename.ext"
=> user_config_path = "/path/to/current/shared/hidra_asapo_plugin.yaml"
```